tests/test_templatetags_slot_fill.py

```diff
@@ -481,98 +698,297 @@ def test_component_without_default_slot_refuses_implicit_fill(self):
+        self.assertHTMLEqual(
+            rendered,
+            """
+            <custom-template>
+                <header>
+                    OVERRIDEN_SLOT "header" - INDEX 0 - ORIGINAL "Default header"
+                </header>
+                <main>
+                    OVERRIDEN_SLOT "main" - INDEX 1 - ORIGINAL "Default main"
+                </main>
+                <footer>
+                    Default footer
+                </footer>
+            </custom-template>
+            """,
+        )
+
+    @parametrize_context_behavior(["isolated", "django"])
+    def test_with(self):
+        @register("test")
+        class SlottedComponent(Component):
+            template: types.django_html = """
+                {% load component_tags %}
+                <custom-template>
+                    <header>{% slot "header" %}Default header{% endslot %}</header>
+                    <main>{% slot "main" %}Default main{% endslot %}</main>
+                    <footer>{% slot "footer" %}Default footer{% endslot %}</footer>
+                </custom-template>
+            """
+
+            def get_context_data(self, name: Optional[str] = None) -> Dict[str, Any]:
+                return {
+                    "name": name,
+                }
 
         template_str: types.django_html = """
             {% load component_tags %}
-            {% component "test1" %}
-                {% fill "haeder" %}
-                    Custom header
-                {% endfill %}
-                {% fill "main" %}
-                    main content
-                {% endfill %}
+            {% component "test" %}
+                {% with slot="header" %}
+                    {% fill name=slot default="default" %}
+                        OVERRIDEN_SLOT "{{ slot }}" - ORIGINAL "{{ default }}"
+                    {% endfill %}
+                {% endwith %}
             {% endcomponent %}
         """
         template = Template(template_str)
 
-        with self.assertRaisesMessage(
-            TemplateSyntaxError,
-            (
-                "Component 'test1' passed fill that refers to undefined slot: 'haeder'.\\n"
-                "Unfilled slot names are: ['footer', 'header'].\\n"
-                "Did you mean 'header'?"
-            ),
-        ):
-            template.render(Context({}))
+        rendered = template.render(Context())
+        self.assertHTMLEqual(
+            rendered,
+            """
+            <custom-template>
+                <header>
+                    OVERRIDEN_SLOT "header" - ORIGINAL "Default header"
+                </header>
+                <main>Default main</main>
+                <footer>Default footer</footer>
+            </custom-template>
+            """,
+        )
 
-    # NOTE: This is relevant only for the "isolated" mode
-    @parametrize_context_behavior(["isolated"])
-    def test_slots_of_top_level_comps_can_access_full_outer_ctx(self):
```

Comment:
This test was moved to here https://github.com/EmilStenstrom/django-components/pull/758/files#diff-ea7f91c03e3b214aa30b65d58c3ea193c51868981b298410097f11e6f457fa01R263